tests/integration/targets/route53_ksk/tasks/main.yml

```diff
@@ -0,0 +1,258 @@
+        key_management_service_arn: "{{ kms_key.key_arn }}"
+        caller_reference: "{{ aws_caller_info.arn }}"
+        status: "INACTIVE"
+        state: present
+      check_mode: true
+      register: _ksk_request
+
+    - name: Assert success
+      ansible.builtin.assert:
+        that:
+          - _ksk_request is successful
+          - _ksk_request.changed
+          - '"msg" in _ksk_request'
+          - '"Would have created the Key Signing Key if not in check_mode." in _ksk_request.msg'
+
+    - name: Create a Key Signing Key request
+      amazon.aws.route53_ksk:
+        name: "{{ resource_prefix }}-ksk"
+        hosted_zone_id: "{{ _hosted_zone.zone_id }}"
+        key_management_service_arn: "{{ kms_key.key_arn }}"
+        caller_reference: "{{ aws_caller_info.arn }}"
+        status: "INACTIVE"
+        state: present
+      register: _ksk_request
+
+    - name: Assert success
+      ansible.builtin.assert:
+        that:
+          - _ksk_request is successful
+          - '"change_info" in _ksk_request'
+          - _ksk_request.changed
+          - '"key_signing_key" in _ksk_request'
+
+    - name: Create a Key Signing Key request (idempotency)
+      amazon.aws.route53_ksk:
+        name: "{{ resource_prefix }}-ksk"
+        hosted_zone_id: "{{ _hosted_zone.zone_id }}"
+        key_management_service_arn: "{{ kms_key.key_arn }}"
+        caller_reference: "{{ aws_caller_info.arn }}"
+        status: "INACTIVE"
+        state: present
+      register: _ksk_request
+
+    - name: Assert success
+      ansible.builtin.assert:
+        that:
+          - _ksk_request is successful
+          - not _ksk_request.changed
+          - '"key_signing_key" in _ksk_request'
+
+    - name: Activate the Key Signing Key request (check_mode)
+      amazon.aws.route53_ksk:
+        name: "{{ resource_prefix }}-ksk"
+        hosted_zone_id: "{{ _hosted_zone.zone_id }}"
+        key_management_service_arn: "{{ kms_key.key_arn }}"
+        caller_reference: "{{ aws_caller_info.arn }}"
+        status: "ACTIVE"
+        state: present
+      check_mode: true
+      register: _ksk_request
+
+    - name: Assert success
+      ansible.builtin.assert:
+        that:
+          - _ksk_request is successful
+          - _ksk_request.changed
+          - '"msg" in _ksk_request'
+          - '"Would have updated the Key Signing Key status to ACTIVE if not in check_mode." in _ksk_request.msg'
+
+    - name: Activate the Key Signing Key request
+      amazon.aws.route53_ksk:
+        name: "{{ resource_prefix }}-ksk"
+        hosted_zone_id: "{{ _hosted_zone.zone_id }}"
+        key_management_service_arn: "{{ kms_key.key_arn }}"
+        caller_reference: "{{ aws_caller_info.arn }}"
+        status: "ACTIVE"
+        state: present
+        wait: true
+      register: _ksk_request
+
+    - name: Assert success
+      ansible.builtin.assert:
+        that:
+          - _ksk_request is successful
+          - '"change_info" in _ksk_request'
```

Comment:
can we check if the status is ACTIVE?

Reply:
This is not possible because `get_change` returns the status of the request. There is no get_* functionality in the API that directly provides information about the KSK. I added a check for the status of the request instead.